dbms/src/Common/SyncPoint/Ctl.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/Exception.h>
+#include <Common/SyncPoint/Ctl.h>
+#include <Common/SyncPoint/ScopeGuard.h>
+#include <Common/SyncPoint/SyncChannel.h>
+#include <common/logger_useful.h>
+#include <fiu-control.h>
+
+namespace DB
+{
+
+#ifdef FIU_ENABLE
+
+void SyncPointCtl::enable(const char * name)
+{
+    {
+        std::unique_lock lock(mu);
+        channels.try_emplace(name,
+                             std::make_pair(
+                                 std::make_shared<SyncChannel>(),
+                                 std::make_shared<SyncChannel>()));
+    }
+    fiu_enable(name, 1, nullptr, 0);
```

Comment:
Does the sync point depends on fiu? I am not sure the reason why we need to call fiu_enable here and fiu_disable in later codes

Reply:
Yes, the sync point depends on the fail point to inject at runtime. When fail point is not enabled by conditional compiling instructions, sync point will be also disabled and has no cost.

Reply:
OK, I get it.